### PR TITLE
[dns_sd] Send an EDNS0 query by default

### PR DESCRIPTION
**Pull request by mpalmer**
Based on https://groups.google.com/d/topic/prometheus-users/02kezHbuea4/discussion

Does not attempt to handle a situation where the server does not understand
EDNS0, however that is an unlikely case, and the behaviour of such ancient
systems is hard to predict in advance, so if it does come up, it will need
to be handled on a case-by-case basis.

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
@SuperQ PTAL

**Comment by mpalmer** [comment]
Incidentally, I can't see what Codacy is complaining about because I get bounced to a login page.

File: discovery/dns/dns.go [hunk 1]
```diff
@@ -284,11 +284,10 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	if err == dns.ErrTruncated {
 		if client.Net == "tcp" {
 			return nil, fmt.Errorf("got truncated message on TCP (64kiB limit exceeded?)")
-		}
-		if edns { // Truncated even though EDNS is used
+		} else {
```
Comment (SuperQ) [inline discovery/dns/dns.go]: Codacy says to drop the `else` and outdent the block contents.  This makes sense to me.

**Comment by mpalmer** [comment]
Fixed the complaint, and also took the opportunity to rebase against current master.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**